public/js/components/SourceSearch.js

```diff
@@ -61,27 +57,27 @@ const Search = React.createClass({
 
   toggle(key, e) {
     e.preventDefault();
-    this.setState({ searchOn: !this.state.searchOn });
+    this.props.toggleSearch(!this.props.searchOn);
   },
 
   onEscape(shortcut, e) {
-    if (this.state.searchOn) {
+    if (this.props.searchOn) {
       e.preventDefault();
-      this.setState({ searchOn: false });
+      this.props.toggleSearch(false);
     }
   },
 
   close() {
-    this.setState({ searchOn: false });
+    this.props.toggleSearch(false);
   },
 
   render() {
-    return this.state.searchOn ?
+    return this.props.searchOn ?
       dom.div({ className: "search-container" },
       Autocomplete({
         selectItem: result => {
           this.props.selectSource(result.id);
-          this.setState({ searchOn: false });
+          this.props.toggleSearch(false);
```

Comment:
I'm not really a fan of having a function that will be bound to an `onClick` handler that then fires a `keydown` event. Maybe this is something that should live in a Redux reducer.